### ISSUE TITLE
fbx model support 

### ISSUE THREAD

**Issue by Chrisc78** [alloystorm/dvvr#383]
I noticed there wasn't a feature request for this yet.
I browse the online store Booth for new .pmx models but most of the models on the website are .fbx
I think DanceXR would get very popular if it starts supporting .fbx models since for now you have to use vrchat to view .fbx models now in VR

A couple really good things about .fbx models, 
1. many high quality models already made and more models are coming out all the time.
2. You can change the clothes of the models using unity, its so easy I can even do it. Most .fbx models on booth has dozens of outfits you can buy.

**Comment by alloystorm** [comment]
Yes this is on the radar for some time but fbx is a complex proprietary format so the only way is to use the official sdk and we haven't found a simple unity integration yet. It will take some time.

**Comment by Chrisc78** [comment]
If the hip broken issue is what I think it is try the model after you delete the hip bone. If there are any Rigidbodys connected to the hip bone you would have to connect them to a different bone first.

**Comment by Chrisc78** [comment]
Whats the bowlroll password for VRM2PmxConverter_v0.5 ?

**Comment by Chrisc78** [comment]
I can't try it until tonight but most purchased models comes with both .fbx and .vrm so maybe  i wouldn't have to do that conversation.

**Comment by Chrisc78** [comment]
> > If the hip broken issue is what I think it is try the model after you delete the hip bone. If there are any Rigidbodys connected to the hip bone you would have to connect them to a different bone first.
> 
> I don't know what exactly it did, but the tool at the link above(it is a plugin for PMX Editor) with the setting "腰ボーンの挙動を修正" and the sub settings ウエイト変更式, 腰キャンセル追加 fixed that issue. Now the model moves as expected (as far as I can tell by eye from a little bit of testing), and, basically, fully works.

Can you take screen shots of how to use the tool?

**Comment by Chrisc78** [comment]
The  VRM2PmxConverter_v0.5 works great, I couldn't get the plug-in tool for pmx editor to work but deleting  the hip bone works perfect. Just make sure to attach any rigidbodys connected to it to a different bone.

**Comment by Chrisc78** [comment]
![Screenshot 2025-01-04 144918](https://github.com/user-attachments/assets/62c3bf1b-1807-49c2-8cb8-3d2287803ce9)

Thanks for letting me know about VRM2PmxConverter_v0.5, I can now change cloths of vrm models and use them in DanceXR. I can't figure out how to setup and convert fbx models but vrm models is easy.

**Comment by Chrisc78** [comment]
> For fbx, I just use .unitypackage directly instead(.fbx is inside there) and then convert it to vrm in Unity, then, after converting it to pmx, here, I use XPS physics.

![Screenshot 2025-01-04 202143](https://github.com/user-attachments/assets/f5856d8e-0a58-48b2-a675-3758ed0e61e4)


Whenever I create a PMX that was once a fbx the face parts detach in DanceXR

**Comment by Chrisc78** [comment]
> Is the .vrm from .fbx ok though? I use UniVRM, and export to vrm 0.x. I have a fully working Kikyo, so you can make sure that you did what I did.
> 
> 1. Import the model's .unitypackage in Unity
> 2. Open the .prefab, select the character
> 3. Export to vrm 0.x through UniVRM
> 4. VRM2PMXConverter 0.5
> 5. Should work in DanceXR
> 
> If it doesn't work, we can check if the versions of Unity and plugins are the same.

It did not work I am using Unity 2021.3.37f1 and UniVRM-0.121.0_640f.unitypackage

**Comment by Chrisc78** [comment]
Something is broken on my end, I uninstalled everything I had then I installed your version, when I load any models unitypackage the textures are hidden and scrips fail to load.

![Screenshot 2025-01-04 225837](https://github.com/user-attachments/assets/4642e23c-68cc-4d60-813c-b2e9dfa8d76e)

**Comment by Chrisc78** [comment]
whats lilToon shader also am I suppose to install any of the following.

![Screenshot 2025-01-04 233155](https://github.com/user-attachments/assets/6b4db4a2-5b50-47bd-a06a-89abe282d00a)

**Comment by Chrisc78** [comment]
That loaded the texures but Kikyo_DB prefab still says missing scripes but Kikyo_PB does not

![Screenshot 2025-01-04 234545](https://github.com/user-attachments/assets/b434bd68-2e8e-4a5a-adef-73113489623c)
![Screenshot 2025-01-04 234617](https://github.com/user-attachments/assets/5f1b6ff8-ecd3-4ee7-a977-6218facd04f9)

**Comment by Chrisc78** [comment]
Thank you for helping me, both prefabs Kikyo_DB and Kikyo_PB worked.

![Screenshot 2025-01-05 000318](https://github.com/user-attachments/assets/c9f3ed33-119e-4ec7-94e5-a064f2c5ce94)

**Comment by Chrisc78** [comment]
This is awesome, I got the pmx Editor plug-in working.

![Screenshot 2025-01-05 013708](https://github.com/user-attachments/assets/875f2527-c449-4ac1-a7ce-a6c80988f42c)

**Comment by Chrisc78** [comment]
I fixed it manually all I did was for each material part was make many copies of the png picture with different names, it might have been easier to do it in pmx editor. Lol

**Comment by Chrisc78** [comment]
The PMX editor plugin can still cause major clipping, the model on the right I deleted the hip bone. I also posted how I do it.
![Screenshot 2025-01-05 150953](https://github.com/user-attachments/assets/5af22d13-3d8c-49d9-b645-296d35cc8bd4)

Delete Bone #23 after you confirm it's not attached to a ridgidbodie
![Screenshot 2025-01-05 151537](https://github.com/user-attachments/assets/12b5e4e5-d156-490f-9c09-a003a348074d)

**Make sure you swap ridgidbodies of from bones you delete I changed bone #23 to bone #24**
![Screenshot 2025-01-05 151934](https://github.com/user-attachments/assets/cd829246-aea1-443e-821f-46e7a433714e)

**Comment by Chrisc78** [comment]
After I rebooted my computer all the script not found errors on the Kikyo model went away.

**Comment by Chrisc78** [comment]
I know the physics is kind of crude but this is my first attempt at setting physics on a vrm

https://github.com/user-attachments/assets/06a680c9-d122-440b-81a9-97a22c525fd8

**Comment by alloystorm** [comment]
Closing this since FBX support is delivered in 2025.9. 
Feel free to open new issues for further improvements and requests.
